File: src/test/scala/com/scala/scogen/datagenerator/DeptDetailsTest.scala
package com.scala.scogen.datagenerator

import com.scala.scogen.BaseClassTest
import org.junit.Assert

class DeptDetailsTest extends BaseClassTest {

  private val deptDataset = new DeptDetailsDataset().deptDataDataset(spark).cache()
  import spark.implicits._

  "count all" should "return all 1000 rows" in {
    Assert.assertEquals(5, deptDataset.count)
  }

  "IT dept " should "exists " in {
    Assert.assertEquals(1, deptDataset.filter($"deptid" === "IT").count)
  }

  "MGR as a dept" should "not be there yet" in{
    Assert.assertEquals(0, deptDataset.filter($"deptid" === "MGR").count)
  }

}

File: src/test/java/com/java/scogen/datagenerator/DeptDetailsTest.java
package com.java.scogen.datagenerator;

import com.java.scogen.SparkTestSessionGenerator;
import org.apache.spark.sql.Column;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.junit.Assert;
import org.junit.Test;

public class DeptDetailsTest extends SparkTestSessionGenerator
{

    private Dataset<Row> deptDataset = new DeptDetailsDataset(getSpark()).deptDataDataset().cache();

    @Test
    public void countTest()
    {
        Assert.assertEquals(5,deptDataset.count());
    }

    @Test
    public void iTDeptExistTest()
    {
        Assert.assertEquals(1,deptDataset.filter(new Column("deptid").equalTo("IT")).count());
    }

    @Test
    public void mgrDeptDpesNotExistTest()
    {
        Assert.assertEquals(0,deptDataset.filter(new Column("deptid").equalTo("MGR")).count());
    }
}
